### PR TITLE
feat(slug): Prevent numeric sentry-doc integration slugs

### PR DESCRIPTION
**Pull request by schew2381**
Prevent numeric DocIntegration slugs. Currently, there is no way to update the slug with a PUT request but there are no existing numeric slugs for this model so we don't need to manually update anything.

### CODE REVIEW

File: src/sentry/api/endpoints/integrations/doc_integrations/index.py [hunk 1]
```diff
@@ -40,12 +40,13 @@ def post(self, request: Request):
...
-        return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=status.HTTP_400_BAD_REQUEST)
+
+        doc_integration = serializer.save()
+        return Response(
+            serialize(doc_integration, request.user),
+            status=status.HTTP_201_CREATED,
+        )
```
Comment (schew2381) [inline src/sentry/api/endpoints/integrations/doc_integrations/index.py:52]: nit: change to early-return

File: src/sentry/api/serializers/rest_framework/doc_integration.py [hunk 2]
```diff
@@ -59,6 +62,12 @@ def validate_name(self, value: str) -> str:
 
     def create(self, validated_data: MutableMapping[str, Any]) -> DocIntegration:
         slug = self._generate_slug(validated_data["name"])
+
+        # If option is set, add random 3 lowercase letter suffix to prevent numeric slug
+        # eg: 123 -> 123-abc
+        if options.get("api.prevent-numeric-slugs") and slug.isnumeric():
```
Comment (cathteng) [inline src/sentry/api/serializers/rest_framework/doc_integration.py:68]:
```suggestion
        if options.get("api.prevent-numeric-slugs") and slug.isdecimal():
```

File: tests/sentry/api/endpoints/test_doc_integrations.py [hunk 3]
```diff
@@ -130,6 +131,20 @@ def test_create_repeated_slug(self):
...
+        Tests that generated slug based on name is not entirely numeric
+        """
+        self.login_as(user=self.superuser, superuser=True)
+        payload = {**self.payload, "name": "1234"}
+        response = self.get_success_response(status_code=status.HTTP_201_CREATED, **payload)
+
+        slug = response.data["slug"]
+        assert len(slug) == 8
+        assert not slug.isnumeric()
+        assert slug.startswith("1234-")
```
Comment (cathteng) [inline tests/sentry/api/endpoints/test_doc_integrations.py:146]:
```suggestion
        assert slug.startswith("1234-")
        assert not slug.isdecimal()
```

**Review by cathteng — APPROVED** [review]
looks good except i think you're switching to `isdecimal()` now

**Comment by schew2381** [comment]
> looks good except i think you're switching to `isdecimal()` now

Nice catch

**Pull request merged**